share/ext/luajit/version.sh

```diff
@@ -15,7 +15,7 @@
 
 
 
-export SHA1=04dca7911ea255f37be799c18d74c305b921c1a6
+export SHA1=a4f56a459a588ae768801074b46ba0adcfb49eb1
```

Comment:
Updated using https://github.com/defold/LuaJIT

Reply:
Did we really have to update LuaJIT?
Wasn't it enough to just rebuild the library?
In general, I'd keep such updates separate, so it's easier to test if anything goes wrong.

Reply:
yes =/
![CleanShot 2025-01-20 at 11 13 48@2x](https://github.com/user-attachments/assets/b6d5e663-4204-4217-ba9b-b1ab75d38cd5)